processor/src/main/java/org/mapstruct/ap/internal/util/AnnotationProcessorContext.java

```diff
@@ -106,20 +107,55 @@ else if ( elementUtils.getTypeElement( FreeBuilderConstants.FREE_BUILDER_FQN ) !
         this.initialized = true;
     }
 
-    private static List<AstModifyingAnnotationProcessor> findAstModifyingAnnotationProcessors() {
+    protected List<AstModifyingAnnotationProcessor> findAstModifyingAnnotationProcessors() {
         List<AstModifyingAnnotationProcessor> processors = new ArrayList<>();
 
         ServiceLoader<AstModifyingAnnotationProcessor> loader = ServiceLoader.load(
                 AstModifyingAnnotationProcessor.class, AnnotationProcessorContext.class.getClassLoader()
         );
 
-        for ( Iterator<AstModifyingAnnotationProcessor> it = loader.iterator(); it.hasNext(); ) {
+        for ( Iterator<AstModifyingAnnotationProcessor> it = loader.iterator(); hasMoreAnnotationProcessors( it ); ) {
             processors.add( it.next() );
         }
 
         return processors;
     }
 
+    protected boolean hasMoreAnnotationProcessors(Iterator<AstModifyingAnnotationProcessor> it) {
+        try {
+            return it.hasNext();
+        }
+        catch ( Throwable t ) {
+            if ( messager != null ) {
+                messager.printMessage( Kind.ERROR,
```

Comment:
I guess this should be a note or warning, since you retry.. Right?

Reply:
Hmm... I'm not retrying with the same element in the iterator. At this point there's at least one error with an element. I'm going to try to skip it, but it's not guaranteed. If trying to skip it fails, then there's another error message warning that some other elements might have failed.